### PR TITLE
test hierarchical gscan

### PR DESCRIPTION
**Pull request by oliver-sanders**
Add a simple test for #931

### CODE REVIEW

**Review by datamel — APPROVED** [review]
I have run the test locally, passed for me.

File: tests/unit/components/cylc/gscan/gscan.vue.spec.js [hunk 1]
```diff
@@ -110,8 +110,24 @@ describe('GScan component', () => {
...
+      while (stack.length) {
+        snode = stack.pop()
+        if (snode && snode.children) {
+          for (const child of snode.children) {
+            ret.push(child.name)
+            stack.push(child)
+          }
+        }
+      }
+      expect(ret).to.deep.equal(['a', 'b', 'c'])
```
Comment (MetRonnie) [inline tests/unit/components/cylc/gscan/gscan.vue.spec.js:128]: Simpler to do?
```suggestion
      expect(nodeHierarchy).to.deep.nested.include({
        name: 'a',
        'children[0].name': 'b',
        'children[0].children[0].name': 'c'
      })
```

**Review by MetRonnie — COMMENTED** [review]
I tested this on 8dd7e52b9d9bc1d7ac355952a4f1baae54f511d7 (commit before the bug was fixed) and it passed 🤔

I have suggested a simplification that passes on master but not on 8dd7e52b9d9bc1d7ac355952a4f1baae54f511d7

File: tests/unit/components/cylc/gscan/gscan.vue.spec.js (hunk 1 above)
Comment (MetRonnie) [reply tests/unit/components/cylc/gscan/gscan.vue.spec.js:128]: It's basically testing for the subset
```js
{
    name: 'a',
    children: [{
        name: 'b',
        children: [{
            name: 'c'
        }]
    }]
}
```
in the object.

Whereas, before #931, it looked like
```js
{
    name: 'a',
    children: [{
        name: 'b',
        children: []
    }, {
        name: 'c',
    }]
}
```
I don't think there should be more than 1 child per level in this function.

I suppose we would also need to check that the `c` node does not have a `children` property

**Pull request closed without merging**